src/shared/browserCaching.ts

```diff
@@ -78,6 +101,13 @@ export function specimenQuery(
 export function parseSpecimenQuery(query: string) {
     const params = new URLSearchParams(query)
     const name = params.get('name') || 'Error: Unknown Name'
+    // We never insert a frame count in queries we generate, but
+    // we do parse it out in case it was specified, e.g. to make
+    // tests reproducible
+    const frames = parseFloat(params.get('frames') || 'Infinity')
+    // Similarly, we never insert a seed, but parse it in case it
+    // was specified
+    const seed = params.get('randomSeed') || null
```

Comment:
So this has the effect that the user can set &frames= or &randomSeed= directly in the URL and get the desired behaviour but as soon as they reload via topbar button or change parameters in the visualizer, that aspect of the URL disappears -- *but is still in force*.  I feel like that might be confusing.  So maybe I set something to &frames=4 and then when I reload or change a parameter it's always drawing only four frames, but there's no way to know that (it's not in the URL).  I know this is kind of a secret ninja functionality that doesn't show up on the parameter pane, but maybe once it is part of the URL it should stay there?

Reply:
In fact, the persistence is not uniform between &randomSeed and &frames.  Right now, if I open up turtle and random sequence, and set a seed via the URL, then the reload button preserves the seed (while eliminating it from the URL), but changing the "maximum value attainable" on the random sequence seems to obliterate the seed (i.e. when I go back to the same parameters the picture has changed).  Whereas both actions appear to preserve the frame limit.

Reply:
A comment that may help resolve what to do about this, if anything:  Sometime in a future PR, I can imagine a use case where I'm wanting to share a randomly generated picture to a friend.  It would be nice to be able to save the seed (whatever it may have been) that was used and have that appear in the URL for sharing.

Reply:
OK, I agree that all this behavior was a bit wonky. The latest commit makes the following changes: (a) Numberscope will not change the URL you _just loaded_; hence raw browser reloads will preserve the seed/frames; (b) changing parameters will remove the frames/seed parameters, as presumably they were for that specific instance of a visualization. So in particular, setting frames to something in the URL and then changing a parameter will then run indefinitely. The only slightly odd case is that setting a randomSeed, and then deleting (say) the "maximum value attainable" and typing back in the _same_ "maximum value attainable" will appear to generate values with the same random seed even though it will now run without a frame limit (and neither the frame limit nor the seed will appear in the URL). What's going on here is that Numberscope is clever enough to notice that the parameter hasn't actually changed, and so it doesn't need to flush the sequence value cache, and so it's just using the random values from the cache that were generated when the random seed was set. You can prove this by changing the max-value-attainable to something else then setting it back to what it was and seeing that now the turtle's path is completely different.

Is that enough of an edge case that we don't need to worry about it? Or is it important that if the randomSeed parameter has disappeared from the URL, then the path **has** to be some fresh new random thing, even if the visitor actually restored all of the parameters to their previous values that were used when the randomSeed had been set? Please take a look at how it is behaving now, and let me know if you think it is OK or if any further changes are needed. Thanks!

Reply:
Hmm, this is a corner case, so I'm ok with filing an issue in order to finish this PR.  And maybe that's the right thing to do, because in the long run I'd like a way to force re-randomize without changing parameters (although we have a lot of buttons already).  So maybe mention this conversation as an issue for if/when we add more user control of the random seed.  But given that this is essentially an internal functionality right now, I don't see why it should slow down this PR.  It is fine as it is.

Reply:
OK, I have filed #461. Does that cover the considerations here appropriately?

Reply:
looks great, thanks